### ISSUE TITLE
Zeros at the end of a number are not recorded in LINDAS

### ISSUE THREAD

**Issue by ortnever** [zazuko/cube-creator#1336]
Zeros at the end of a number are not recorded in LINDAS, whereas they are imported correctly in the cube creator and appear in the Cube Designer.

It is important that these zeros at the end of a number are saved, because with the number of decimal places we also make a statement about the "uncertainty". 

Is this something that could be corrected quickly?

For the tests, you can use the following cube: https://environment.ld.admin.ch/foen/ubd000502_sad_01/7 on Lindas INT.

![image](https://user-images.githubusercontent.com/79834090/203511637-1a3e9d0b-1090-4eea-8c72-53d72d0ba2cd.png)

![image](https://user-images.githubusercontent.com/79834090/203511615-7909aa9e-2602-4066-9409-86568dbbdf01.png)

**Comment by tpluscode** [comment]
> a `format` annotation to use by Visualize would be the most appropriate course of action

ping @ptbrowne
What do you think about a display format annotation on dimension metadata?

**Comment by ptbrowne** [comment]
I think display can change according to the need of the users. I think what should be in the data should be the **resolution** at which the data has been recorded, and then in the UI we could choose the formatting.

Right now in visualize, we use the concept of resolution to format integers without a comma. This resolution variable could be taken dynamically from the dimension and passed to the front-end so that it can make appropriate choices.

I think it'd be important to choose the right vocabulary here : https://www.crossco.com/resources/articles/metrology-terminology-accuracy-precision-resolution/, I would have thought that qudt would provide us with this kind of vocabulary, I took a quick look but could not find anything.

Also related is how currency are formatted : [Docs](https://int.visualize.admin.ch/docs/charts/rdf-to-visualize#currencies), [Code](https://github.com/visualize-admin/visualization-tool/blob/9261565b05db4b2bc0c79a1fe2c0f35c1142ee14/app/formatters.ts#L46)

**Comment by l00mi** [comment]
compare to https://gitlab.ldbar.ch/bafu/umweltdatenkiosk-planning/-/issues/332#note_9360

**Comment by l00mi** [comment]
So I had a talk to a machine:

> how do your refer to the display of the amount of zeros after the comma of a number ?

The number of zeros after the comma (or period) of a number is commonly referred to as the "number of decimal places". For example, in the number 3.14159, there are five decimal places.
Alternatively, the number of decimal places can also be referred to as the "precision" or "significant figures" of the number. These terms refer to the level of detail or accuracy that the number represents. For example, a number with three decimal places (e.g. 3.141) implies a higher level of precision than a number with only one decimal place (e.g. 3.1).
It's important to note that the number of decimal places, precision, and significant figures all refer to the same concept, but they may be used in different contexts or with different conventions.

> Is it the same a qudt:precision ?

No, QUDT (Quantities, Units, Dimensions, and Data Types Ontology) is a standardized ontology for describing physical quantities, units of measurement, and their relationships. The QUDT specification includes a property called "qudt:precision", which is used to represent the number of significant figures or decimal places in a measured value.
QUDT's "qudt:precision" property is similar in meaning to the concept of precision discussed earlier, but it is a specific property defined within the QUDT ontology, which is designed to provide a consistent way to represent precision in measurements within a broader semantic framework.
In summary, while both "precision" and "qudt:precision" relate to the level of detail or accuracy in a measurement, "qudt:precision" is a specific property defined within the QUDT ontology to represent this information in a standardized way.

That would be an example of its use:
```
ex:LengthMeasurement rdf:type qudt:QuantityValue ;
    qudt:numericValue "5.372"^^xsd:decimal ;
    qudt:unit ex:Meter ;
    qudt:precision "3"^^xsd:nonNegativeInteger .
```
While the machine was very vague, I guess it is save to use qudt:precision for this use case. Where the number is the amount of digits after the coma.

**Comment by l00mi** [comment]
@ptbrowne so its now up to you to say that this will be enough, or shall we go further and try something more power full in the regards of good ol 'sprintf()' .. e.g. https://locutus.io/php/strings/sprintf/

**Comment by ortnever** [comment]
@ptbrowne : I add an example of a use case below, because I don't understand the whole discussion and I would like to be sure that it meets our needs.

Columns A to E are the delivered data.
We would like that Visualize displays the numbers of column D rounded to the precision indicated in column E.
Column G is of course not part of the source data and is just there to show what we would like Visualize to display.

@Thomas: does this use case address other needs and is it general enough? 

![image](https://user-images.githubusercontent.com/79834090/219615869-2f294686-9eb0-4893-b72b-6dca45d68412.png)

**Comment by ptbrowne** [comment]
Thanks for providing an example Veronique. From my end what I see here is

- The precision metadata should not be on the column metadata but on the observation since observations can have different associated precision
- A new type of relation between dimensions should be specified, like for standard errors
- Given those relations, visualize could format the data according to your needs

**Comment by ortnever** [comment]
exactly. But it's only one use case. Maybe there are other ones in relation with this issue.

**Comment by Rdataflow** [comment]
I think there are two generic cases as of now:
- define precision per measure dimension (to be stored in the dimension property)
- define precision per value (to be stored as secondary values in `related` dimension)

**Comment by l00mi** [comment]
So the precision per value can be more quickly solved, as we have the structure as for the standard error. 

Best to create an issue for each of the both use cases. And then priorize them individually.
